mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -893,17 +895,25 @@ struct AttentionRewritePattern : public OpRewritePattern<tosa::MatMulOp> {
   getPreSoftmaxElemwiseRegion(Value input, OpBuilder &regionBuilder,
                               Block *block, SmallVector<Value> &elemwiseArgs,
                               std::optional<Location> loc = std::nullopt,
-                              bool doRewrite = false) const {
+                              bool doRewrite = false, int recDepth = 0) const {
     PatternRewriter::InsertionGuard guard(regionBuilder);
     regionBuilder.setInsertionPointToEnd(block);
     // If the matmul is found, we return this information to the
     // root.
+    LLVM_DEBUG(llvm::dbgs()
+               << std::string(recDepth, '\t')
```

Comment:
Why `recDepth`

Reply:
gets a nice log of how recursion went with `-debug-only`